PyInstaller/building/splash.py

```diff
@@ -218,6 +218,24 @@ def __init__(self, image_file, binaries, datas, **kwargs):
             # true.
             self.binaries.extend(toc for toc in tcltk_tree if toc[0] in self.splash_requirements)
 
+        # Handle extra requirements of Tcl/Tk shared libraries (e.g., vcruntime140.dll on Windows - see issue #6284).
+        # These need to be added to splash requirements, so they are extracted into the initial runtime directory in
+        # order to make onefile builds work.
+        #
+        # The really proper way to implement this would be to perform full dependency analysis on self.tcl_lib[0] and
+        # self.tk_lib[0], and ensure that those dependencies are collected and added to splash requirements. This
+        # would, for example, ensure that on Linux, dependent X libraries are collected, just as if the frozen app
+        # itself was using tkinter. On the other hand, collecting all the extra shared libraries on Linux is currently
+        # futile anyway, because the bootloader's parent process would need to set LD_LIBRARY_PATH to the initial
+        # runtime directory to actually have them loaded (and that requires process to be restarted to take effect).
+        #
+        # So for now, we only deal with this on Windows, in a quick'n'dirty work-around way, by assuming that
+        # vcruntime140.dll is already collected as dependency of some other shared library (e.g., the python shared
+        # library).
+        if is_win or is_cygwin:
+            EXTRA_REQUIREMENTS = {'vcruntime140.dll'}
+            self.splash_requirements.update([name for name, *_ in binaries if name.lower() in EXTRA_REQUIREMENTS])
```

Comment:
Could we just conditionally add `ctypes.util.find_library("vcruntime140.dll")` to [`splash_requirements` at the top](https://github.com/pyinstaller/pyinstaller/pull/6668/files#diff-71eda1c9c6afe40eb52184fcbbb13aee550548490f07e9095c63aa63607be6fbR35)? I find that less mind twisting than searching `binaries` for a file whose name we already know and whose presence in that list is only assured because Python happens to also depend on this library.

Reply:
We don't need the full path to the library, only its name. But we probably need its name with correct (lower/upper) case, hence the search for the name. I don't think the relevant comparisons (the filtering of splash_requirements couple in subsequent lines of code, and TOC entry look-up in the bootloader) are done in case-insensitive way...